src/main/groovy/wooga/gradle/version/VersionPlugin.groovy

```diff
@@ -60,13 +60,10 @@ class VersionPlugin implements Plugin<Project> {
         }
         finally {
             def sharedVersion = new ToStringProvider(extension.version.map({ it.version }))
-            project.allprojects(new Action<Project>() {
-                @Override
-                void execute(Project prj) {
-                    prj.setVersion(sharedVersion)
-                    prj.extensions.create(VersionCodeExtension, "versionCode", DefaultVersionCodeExtension.class, prj, extension.versionCode)
-                }
-            })
+            project.allprojects{ Project prj ->
+                prj.setVersion(sharedVersion)
+                prj.extensions.create(VersionCodeExtension, "versionCode", DefaultVersionCodeExtension.class, prj, extension.versionCode)
+            }
```

Comment:
These kind of refactors feel unnessary. We can discuss in generel if we should use the java or groovy API. I used the java API in most cases since the short groovy version is not always recognized by IDEA. Also the code is not formated correctly here as well.

Reply:
This doesn't uses groovy API, the Action class is a functional interface and so groovy 2 supports using the closure as a Lambda. Also, I believe this is much more readable than the Anonymous Class alternative. Even in Java 8 this kind of stuff would be written using a lambda (`project.allprojects(prj -> { //method })`) instead of an explicit Anonymous class. But groovy 2 doesn't support java 8 lambda syntax.

Closures probably have some extra overhead compared with lambdas, but I don't think this is relevant, if we were worried about this kind of small overhead we should just stop using groovy, as it adds a lot of calls everywhere.

Finally, I had no problems with Idea recognizing Closures in those contexts so far. Maybe its a problem on older versions?

Reply:
It is not about that is more readable. You changed it here for no other reason. It is functional the same. I know that in this PR it is only this one place. But the context of this PR is to refactor a specific piece and not change the code style as well. Leave it for this PR but try to refain from changing codestyle along with other changes. Also this is no lambda. this is basic groovy closures. The API of gradle exposes most methods with the Action interface or Groovy Closure API. You just switched from using the former to the later.

Reply:
Well, yeah, just double checked here and the API indeed have a Closure parameter as well, but the effect of the action ends up being the same, just with some extra gradle steps.

I know it's harder to review when a PR have a bunch of aesthetic changes that aren't directly related to its content, but I do believe that it's always good to improve code when there is an opportunity to do so. Those continuous small improvements are good on combating code decay on general, in my experience. 

However I do agree that the PR system is not very conducive for that, and that this kind of change works better on its own PR. I can understand where you are coming from and I'll try to not do this kind of change mixed on PRs anymore, but I think that we are missing something with it.